Fails to compile for macOS (Apple Silicon)

M1 iMac
macOS 13.2.1
Xcode 14.3

Haven't dove into the code, but compilation for macOS arm64 yields the following log excerpt:

```
SwiftCompile normal arm64 [PATH]/Map/Sources/Map/Map.swift (in target 'Map' from project 'Map')

...

[PATH]/Map/Sources/Map/Map.swift:55:139: error: contextual closure type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?' expects 2 arguments, but 1 was used in closure body
        @OptionalMapAnnotationBuilder clusterAnnotation: @escaping (MKClusterAnnotation, [AnnotationItems.Element]) -> MapAnnotation? = { _ in nil },
                                                                                                                                          ^
                                                                                                                                           ,_ 
[PATH]/Map/Sources/Map/Map.swift:55:137: error: generic parameter 'AnnotationItems' could not be inferred
        @OptionalMapAnnotationBuilder clusterAnnotation: @escaping (MKClusterAnnotation, [AnnotationItems.Element]) -> MapAnnotation? = { _ in nil },
                                                                                                                                        ^
[PATH]/Map/Sources/Map/Map.swift:13:19: note: 'AnnotationItems' declared as parameter to type 'Map'
public struct Map<AnnotationItems: RandomAccessCollection, OverlayItems: RandomAccessCollection>
                  ^
[PATH]/Map/Sources/Map/Map.swift:285:32: error: extra argument 'clusterAnnotation' in call
            clusterAnnotation: { annotation, _ in clusterAnnotation(annotation) },
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[PATH]/Map/Sources/Map/Map.swift:493:32: error: cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'
            clusterAnnotation: clusterAnnotation,
                               ^
[PATH]/Map/Sources/Map/Map.swift:493:32: error: cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'
            clusterAnnotation: clusterAnnotation,
                               ^
[PATH]/Map/Sources/Map/Map.swift:524:32: error: cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'
            clusterAnnotation: clusterAnnotation,
                               ^
[PATH]/Map/Sources/Map/Map.swift:524:32: error: cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'
            clusterAnnotation: clusterAnnotation,
                               ^
[PATH]/Map/Sources/Map/Map.swift:558:32: error: cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'
            clusterAnnotation: clusterAnnotation,
                               ^
[PATH]/Map/Sources/Map/Map.swift:558:32: error: cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'
            clusterAnnotation: clusterAnnotation,
                               ^
[PATH]/Map/Sources/Map/Map.swift:592:32: error: cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'
            clusterAnnotation: clusterAnnotation,
                               ^
[PATH]/Map/Sources/Map/Map.swift:592:32: error: cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'
            clusterAnnotation: clusterAnnotation,
                               ^
[PATH]/Map/Sources/Map/Map.swift:711:32: error: contextual closure type '(MKClusterAnnotation, [IdentifiableObject<any MKAnnotation>]) -> (any MapAnnotation)?' expects 2 arguments, but 1 was used in closure body
            clusterAnnotation: { clusterAnnotation($0.map(\.object)) },
                               ^
[PATH]/Map/Sources/Map/Map.swift:711:55: error: value of type 'MKClusterAnnotation' has no member 'map'
            clusterAnnotation: { clusterAnnotation($0.map(\.object)) },
                                                   ~~ ^~~
[PATH]/Map/Sources/Map/Map.swift:711:59: error: cannot infer key path type from context; consider explicitly specifying a root type
            clusterAnnotation: { clusterAnnotation($0.map(\.object)) },
                                                          ^
                                                           <#Root#>
[PATH]/Map/Sources/Map/Map.swift:745:32: error: contextual closure type '(MKClusterAnnotation, [IdentifiableObject<any MKAnnotation>]) -> (any MapAnnotation)?' expects 2 arguments, but 1 was used in closure body
            clusterAnnotation: { clusterAnnotation($0.map(\.object)) },
                               ^
[PATH]/Map/Sources/Map/Map.swift:745:55: error: value of type 'MKClusterAnnotation' has no member 'map'
            clusterAnnotation: { clusterAnnotation($0.map(\.object)) },
                                                   ~~ ^~~
[PATH]/Map/Sources/Map/Map.swift:745:59: error: cannot infer key path type from context; consider explicitly specifying a root type
            clusterAnnotation: { clusterAnnotation($0.map(\.object)) },
                                                          ^
                                                           <#Root#>
[PATH]/Map/Sources/Map/Map.swift:782:32: error: contextual closure type '(MKClusterAnnotation, [IdentifiableObject<any MKAnnotation>]) -> (any MapAnnotation)?' expects 2 arguments, but 1 was used in closure body
            clusterAnnotation: { clusterAnnotation($0.map(\.object)) },
                               ^
[PATH]/Map/Sources/Map/Map.swift:782:55: error: value of type 'MKClusterAnnotation' has no member 'map'
            clusterAnnotation: { clusterAnnotation($0.map(\.object)) },
                                                   ~~ ^~~
[PATH]/Map/Sources/Map/Map.swift:782:59: error: cannot infer key path type from context; consider explicitly specifying a root type
            clusterAnnotation: { clusterAnnotation($0.map(\.object)) },
                                                          ^
                                                           <#Root#>

[PATH]/Map/Sources/Map/Map.swift:55:139: Contextual closure type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?' expects 2 arguments, but 1 was used in closure body

[PATH]/Map/Sources/Map/Map.swift:55:137: Generic parameter 'AnnotationItems' could not be inferred

[PATH]/Map/Sources/Map/Map.swift:285:32: Extra argument 'clusterAnnotation' in call

[PATH]/Map/Sources/Map/Map.swift:493:32: Cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'

[PATH]/Map/Sources/Map/Map.swift:493:32: Cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'

[PATH]/Map/Sources/Map/Map.swift:524:32: Cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'

[PATH]/Map/Sources/Map/Map.swift:524:32: Cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'

[PATH]/Map/Sources/Map/Map.swift:558:32: Cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'

[PATH]/Map/Sources/Map/Map.swift:558:32: Cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'

[PATH]/Map/Sources/Map/Map.swift:592:32: Cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'

[PATH]/Map/Sources/Map/Map.swift:592:32: Cannot convert value of type '([AnnotationItems.Element]) -> (any MapAnnotation)?' to expected argument type '(MKClusterAnnotation, [AnnotationItems.Element]) -> (any MapAnnotation)?'

[PATH]/Map/Sources/Map/Map.swift:711:32: Contextual closure type '(MKClusterAnnotation, [IdentifiableObject<any MKAnnotation>]) -> (any MapAnnotation)?' expects 2 arguments, but 1 was used in closure body

[PATH]/Map/Sources/Map/Map.swift:711:55: Value of type 'MKClusterAnnotation' has no member 'map'

[PATH]/Map/Sources/Map/Map.swift:711:59: Cannot infer key path type from context; consider explicitly specifying a root type

[PATH]/Map/Sources/Map/Map.swift:745:32: Contextual closure type '(MKClusterAnnotation, [IdentifiableObject<any MKAnnotation>]) -> (any MapAnnotation)?' expects 2 arguments, but 1 was used in closure body

[PATH]/Map/Sources/Map/Map.swift:745:55: Value of type 'MKClusterAnnotation' has no member 'map'

[PATH]/Map/Sources/Map/Map.swift:745:59: Cannot infer key path type from context; consider explicitly specifying a root type

[PATH]/Map/Sources/Map/Map.swift:782:32: Contextual closure type '(MKClusterAnnotation, [IdentifiableObject<any MKAnnotation>]) -> (any MapAnnotation)?' expects 2 arguments, but 1 was used in closure body

[PATH]/Map/Sources/Map/Map.swift:782:55: Value of type 'MKClusterAnnotation' has no member 'map'

[PATH]/Map/Sources/Map/Map.swift:782:59: Cannot infer key path type from context; consider explicitly specifying a root type
```

--- a/Sources/Map/Map.swift
+++ b/Sources/Map/Map.swift
@@ -52,7 +52,7 @@ extension Map {
         interactionModes: MapInteractionModes = .all,
         annotationItems: AnnotationItems,
         @MapAnnotationBuilder annotationContent: @escaping (AnnotationItems.Element) -> MapAnnotation,
-        @OptionalMapAnnotationBuilder clusterAnnotation: @escaping (MKClusterAnnotation, [AnnotationItems.Element]) -> MapAnnotation? = { _ in nil },
+        @OptionalMapAnnotationBuilder clusterAnnotation: @escaping (MKClusterAnnotation, [AnnotationItems.Element]) -> MapAnnotation? = { _, _ in nil },
         overlayItems: OverlayItems,
         @MapOverlayBuilder overlayContent: @escaping (OverlayItems.Element) -> MapOverlay
     ) {
@@ -282,7 +282,6 @@ extension Map where AnnotationItems == [IdentifiableObject<MKAnnotation>] {
             annotationItems: annotations.map(IdentifiableObject.init),
             annotationContent: { annotationContent($0.object) },
             clusterAnnotation: { annotation, _ in clusterAnnotation(annotation) },
-            clusterAnnotation: { annotation, _ in clusterAnnotation(annotation) },
             overlayItems: overlayItems,
             overlayContent: overlayContent
         )
@@ -473,7 +472,7 @@ extension Map where OverlayItems == [IdentifiableObject<MKOverlay>] {
         interactionModes: MapInteractionModes = .all,
         annotationItems: AnnotationItems,
         @MapAnnotationBuilder annotationContent: @escaping (AnnotationItems.Element) -> MapAnnotation,
-        @OptionalMapAnnotationBuilder clusterAnnotation: @escaping ([AnnotationItems.Element]) -> MapAnnotation? = { _ in nil },
+        @OptionalMapAnnotationBuilder clusterAnnotation: @escaping (MKClusterAnnotation, [AnnotationItems.Element]) -> MapAnnotation? = { _, _ in nil },
         overlays: [MKOverlay] = [],
         @MapOverlayBuilder overlayContent: @escaping (MKOverlay) -> MapOverlay = { overlay in
             assertionFailure("Please provide an `overlayContent` closure for the values in `overlays`.")
@@ -504,7 +503,7 @@ extension Map where OverlayItems == [IdentifiableObject<MKOverlay>] {
         interactionModes: MapInteractionModes = .all,
         annotationItems: AnnotationItems,
         @MapAnnotationBuilder annotationContent: @escaping (AnnotationItems.Element) -> MapAnnotation,
-        @OptionalMapAnnotationBuilder clusterAnnotation: @escaping ([AnnotationItems.Element]) -> MapAnnotation? = { _ in nil },
+        @OptionalMapAnnotationBuilder clusterAnnotation: @escaping (MKClusterAnnotation, [AnnotationItems.Element]) -> MapAnnotation? = { _, _ in nil },
         overlays: [MKOverlay] = [],
         @MapOverlayBuilder overlayContent: @escaping (MKOverlay) -> MapOverlay = { overlay in
             assertionFailure("Please provide an `overlayContent` closure for the values in `overlays`.")
@@ -537,7 +536,7 @@ extension Map where OverlayItems == [IdentifiableObject<MKOverlay>] {
         userTrackingMode: Binding<UserTrackingMode>?,
         annotationItems: AnnotationItems,
         @MapAnnotationBuilder annotationContent: @escaping (AnnotationItems.Element) -> MapAnnotation,
-        @OptionalMapAnnotationBuilder clusterAnnotation: @escaping ([AnnotationItems.Element]) -> MapAnnotation? = { _ in nil },
+        @OptionalMapAnnotationBuilder clusterAnnotation: @escaping (MKClusterAnnotation, [AnnotationItems.Element]) -> MapAnnotation? = { _, _ in nil },
         overlays: [MKOverlay] = [],
         @MapOverlayBuilder overlayContent: @escaping (MKOverlay) -> MapOverlay = { overlay in
             assertionFailure("Please provide an `overlayContent` closure for the values in `overlays`.")
@@ -571,7 +570,7 @@ extension Map where OverlayItems == [IdentifiableObject<MKOverlay>] {
         userTrackingMode: Binding<UserTrackingMode>?,
         annotationItems: AnnotationItems,
         @MapAnnotationBuilder annotationContent: @escaping (AnnotationItems.Element) -> MapAnnotation,
-        @OptionalMapAnnotationBuilder clusterAnnotation: @escaping ([AnnotationItems.Element]) -> MapAnnotation? = { _ in nil },
+        @OptionalMapAnnotationBuilder clusterAnnotation: @escaping (MKClusterAnnotation, [AnnotationItems.Element]) -> MapAnnotation? = { _, _ in nil },
         overlays: [MKOverlay] = [],
         @MapOverlayBuilder overlayContent: @escaping (MKOverlay) -> MapOverlay = { overlay in
             assertionFailure("Please provide an `overlayContent` closure for the values in `overlays`.")
@@ -708,7 +707,7 @@ extension Map where AnnotationItems == [IdentifiableObject<MKAnnotation>], Overl
             interactionModes: interactionModes,
             annotationItems: annotations.map(IdentifiableObject.init),
             annotationContent: { annotationContent($0.object) },
-            clusterAnnotation: { clusterAnnotation($0.map(\.object)) },
+            clusterAnnotation: { annotation, _ in clusterAnnotation(annotation) },
             overlayItems: overlays.map(IdentifiableObject.init),
             overlayContent: { overlayContent($0.object) }
         )
@@ -742,7 +741,7 @@ extension Map where AnnotationItems == [IdentifiableObject<MKAnnotation>], Overl
             interactionModes: interactionModes,
             annotationItems: annotations.map(IdentifiableObject.init),
             annotationContent: { annotationContent($0.object) },
-            clusterAnnotation: { clusterAnnotation($0.map(\.object)) },
+            clusterAnnotation: { annotation, _ in clusterAnnotation(annotation) },
             overlayItems: overlays.map(IdentifiableObject.init),
             overlayContent: { overlayContent($0.object) }
         )
@@ -779,7 +778,7 @@ extension Map where AnnotationItems == [IdentifiableObject<MKAnnotation>], Overl
             userTrackingMode: userTrackingMode,
             annotationItems: annotations.map(IdentifiableObject.init),
             annotationContent: { annotationContent($0.object) },
-            clusterAnnotation: { clusterAnnotation($0.map(\.object)) },
+            clusterAnnotation: { annotation, _ in clusterAnnotation(annotation) },
             overlayItems: overlays.map(IdentifiableObject.init),
             overlayContent: { overlayContent($0.object) }
         )

